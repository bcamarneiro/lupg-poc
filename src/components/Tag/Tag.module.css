.tag {
  @apply py-1 px-[6px] rounded-full text-xs capitalize;
}
