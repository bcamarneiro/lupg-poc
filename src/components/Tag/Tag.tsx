import { PropsWithChildren } from 'react';

import style from './Tag.module.css';
import clsx from 'clsx';

function Tag({ children }: PropsWithChildren) {
  return <span className={clsx(style.tag, 'bg-levelup-500 dark:bg-levelup-300')}>{children}</span>;
}
export default Tag;
