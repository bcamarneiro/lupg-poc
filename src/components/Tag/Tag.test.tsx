import { render, screen } from '@testing-library/react';
import Tag from './Tag';

describe('Tag component', () => {
  it('renders tag with children correctly', () => {
    const { asFragment } = render(<Tag>Test Tag</Tag>);

    expect(screen.getByText('Test Tag')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom children', () => {
    render(
      <Tag>
        <span>Custom</span>
        <strong>Content</strong>
      </Tag>,
    );

    expect(screen.getByText('Custom')).toBeInTheDocument();
    expect(screen.getByText('Content')).toBeInTheDocument();
  });
});
