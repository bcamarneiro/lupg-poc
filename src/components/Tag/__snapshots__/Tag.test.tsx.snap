// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag component > renders tag with children correctly 1`] = `
<DocumentFragment>
  <span
    class="_tag_f783ef bg-levelup-500 dark:bg-levelup-300"
  >
    Test Tag
  </span>
</DocumentFragment>
`;
